packages/fast-components-styles-msft/src/utilities/density.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { DesignSystem, withDesignSystemDefaults } from "../design-system";
+import {
+    DensityCategory,
+    getDensityCategory,
+    getOffsetForDensityCategory,
+    glyphSize,
+    height,
+    horizontalSpacing,
+} from "./density";
+
+describe("density", (): void => {
+    const spaciousDesignSystem: DesignSystem = withDesignSystemDefaults({
+        density: 2,
+        designUnit: 5,
+        baseHeightMultiplier: 9,
+        baseHorizontalSpacingMultiplier: 4,
+    });
+
+    const compactDesignSystem: DesignSystem = withDesignSystemDefaults({
+        density: -2,
+        designUnit: 3,
+        baseHeightMultiplier: 7,
+        baseHorizontalSpacingMultiplier: 2,
+    });
+
+    test("should operate on design system defaults", (): void => {
+        expect(height()({} as DesignSystem)).toBe("32px");
+        expect(height(2)({} as DesignSystem)).toBe("64px");
+        expect(height(2, "em")({} as DesignSystem)).toBe("64em");
+        expect(getDensityCategory({} as DesignSystem)).toBe(DensityCategory.normal);
+        expect(getOffsetForDensityCategory(-1, 1)({} as DesignSystem)).toBe(0);
+        expect(getOffsetForDensityCategory(3, 2, 1)({} as DesignSystem)).toBe(1);
+        expect(horizontalSpacing()({} as DesignSystem)).toBe("12px");
+        expect(horizontalSpacing(2)({} as DesignSystem)).toBe("10px");
+        expect(horizontalSpacing(2, "em")({} as DesignSystem)).toBe("10em");
+        expect(glyphSize({} as DesignSystem)).toBe("16px");
+        expect(glyphSize("em")({} as DesignSystem)).toBe("16em");
+    });
+
+    test("should operate on spacious design system", (): void => {
+        expect(height()(spaciousDesignSystem)).toBe("55px");
```

Comment:
Should we be checking for exact values? Any change in the design system values will result in having to update all of these tests. I would think that a more flexible way to go is to make sure that adding the spacious design system, compact design system etc would produce a different result than the default design system.

Reply:
I thought about this and used hard coded values because otherwise the tests would just end up duplicating the same calculations in the methods. I suppose if I rewrote the calculation it would probably be a sufficient test though.

Reply:
Updated